packages/jest-cli/src/reporters/coverage_reporter.js

```diff
@@ -125,7 +125,8 @@ export default class CoverageReporter extends BaseReporter {
       const config = context.config;
       if (
         globalConfig.collectCoverageFrom &&
-        globalConfig.collectCoverageFrom.length
+        globalConfig.collectCoverageFrom.length &&
```

Comment:
Should we also check for `!config.watchAll`?

Reply:
I don't think so - the `watchAll` is true when all files are watched in watch mode.
`watch` is true when there is some filtering applied, e.g. by test file name.